### PR TITLE
Replace load_boston() with fetch_california_housing()

### LINKED ISSUE #497: Replace load_boston with alternate regression dataset
**Describe the bug**
As noted in https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_boston.html the Boston housing dataset is being deprecated due to a significant ethical concern:

Warning: The Boston housing prices dataset has an ethical problem: as investigated in [1], the authors of this dataset engineered a non-invertible variable “B” assuming that racial self-segregation had a positive impact on house prices [2]. Furthermore the goal of the research that led to the creation of this dataset was to study the impact of air quality but it did not give adequate demonstration of the validity of this assumption.

The scikit-learn maintainers therefore strongly discourage the use of this dataset unless the purpose of the code is to study and educate about ethical issues in data science and machine learning.

Sklearn suggests that either the California housing dataset or the Ames housing dataset are reasonable alternative datasets [https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_boston.html]. We should change references in interpret-community accordingly.

**To Reproduce**
1. Search for load_boston in interpret-community repository

**Expected behavior**
Sklearn suggests that either the California housing dataset or the Ames housing dataset are reasonable alternative datasets [https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_boston.html]. We should change references in interpret-community accordingly.

### PR DESCRIPTION
**Pull request by gaugup**
Signed-off-by: Gaurav Gupta <gaugup@microsoft.com>

### CODE REVIEW

File: notebooks/explain-regression-local.ipynb [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "<a id='Project'></a>       \n",
     "## 2. Project\n",
     "\n",
-    "The goal of this project is to predict Boston Housing Prices by using scikit-learn and locally running the model explainer:\n",
+    "The goal of this project is to predict housing Housing Prices by using scikit-learn and locally running the model explainer:\n",
```
Comment (imatiach-msft) [inline notebooks/explain-regression-local.ipynb:56]: housing Housing -> California housing

File: tests/test_explain_model.py [hunk 2]
```diff
@@ -906,15 +906,15 @@ def iris_per_class_expected_features(self):
                 ['petal length', 'petal width', 'sepal length', 'sepal width']]
 
     @property
-    def boston_local_features_first_five_rf(self):
+    def housing_local_features_first_five_rf(self):
         return [['LSTAT', 'CRIM', 'B', 'AGE', 'INDUS', 'RAD', 'CHAS', 'ZN', 'TAX', 'DIS', 'PTRATIO', 'NOX', 'RM'],
```
Comment (imatiach-msft) [inline tests/test_explain_model.py:910]: I think all of these checks will need to be updated to the new features

Comment (gaugup) [reply tests/test_explain_model.py]: Done.

File: notebooks/explain-regression-local.ipynb (hunk 1 above)
Comment (gaugup) [reply notebooks/explain-regression-local.ipynb]: Done.

File: tests/test_explain_model.py [hunk 3]
```diff
@@ -728,7 +728,7 @@ def test_explain_model_pandas_string(self, tabular_explainer):
         B = np.random.random(size=num_rows)
         C = np.random.randn(num_rows)
         cat = np.random.choice(['New York', 'San Francisco', 'Los Angeles',
-                                'Atlanta', 'Denver', 'Chicago', 'Miami', 'DC', 'Boston'], 100)
+                                'Atlanta', 'Denver', 'Chicago', 'Miami', 'DC', 'housing'], 100)
```
Comment (imatiach-msft) [inline tests/test_explain_model.py:731]: this looks a bit weird/wrong?

**Review by imatiach-msft — APPROVED** [review]

File: tests/test_explain_model.py (hunk 3 above)
Comment (gaugup) [reply tests/test_explain_model.py]: Reverted it. Thanks for pointing this out.

**Pull request merged**